Library/Homebrew/tap.rb

```diff
@@ -0,0 +1,93 @@
+class Tap
+  TAP_DIRECTORY = HOMEBREW_LIBRARY/"Taps"
+
+  extend Enumerable
+
+  attr_reader :user
+  attr_reader :repo
+  attr_reader :name
+  attr_reader :path
+  attr_reader :remote
+
+  def initialize(user, repo, remote=nil)
+    # we special case homebrew so users don't have to shift in a terminal
+    @user = user == "homebrew" ? "Homebrew" : user
+    @repo = repo
+    @name = "#{@user}/#{@repo}".downcase
+    @path = TAP_DIRECTORY/"#{@user}/homebrew-#{@repo}".downcase
+    if installed?
+      @path.cd do
+        @remote = Utils.popen_read("git", "config", "--get", "remote.origin.url").chomp
+      end
+    else
+      @remote = remote || "https://github.com/#{@user}/homebrew-#{@repo}"
+    end
+  end
+
+  def to_s
+    name
+  end
+
+  def official?
+    @user == "Homebrew"
+  end
+
+  def private?
+    return true if anytap?
+    GitHub.private_repo?(user, "homebrew-#{repo}")
+  rescue GitHub::HTTPNotFoundError
+    true
+  rescue GitHub::Error
+    false
+  end
+
+  def installed?
+    (@path/".git").directory?
+  end
+
+  def anytap?
+    @remote.casecmp("https://github.com/#{user}/homebrew-#{repo}") != 0
+  end
+
+  def formula_files
+    return [] unless installed?
+    files = []
+    @path.find_formula { |file| files << file }
```

Comment:
Is this now the only place where find_formula is used? Can we inline it here? It's always bugged me that it's a method on pathname.